packages/components/pie-modal/src/index.ts

```diff
@@ -359,85 +356,93 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
-
-        const scrollContainerClasses = {
-            'c-modal-scrollContainer': true,
-            'c-modal-content': true,
-            'c-modal-content--scrollable': true,
-            'c-modal-hasFooterActions': hasFooterLeadingAction,
-        };
+    private renderModalFooter () : TemplateResult | typeof nothing {
+        if (!this.leadingAction?.text) {
+            if (this.supportingAction?.text) {
+                console.warn('You cannot have a supporting action without a leading action. If you only need one button then use a leading action instead.');
+            }
+            return nothing;
+        }
 
         return html`
-            <article class=${classMap(scrollContainerClasses)}>
+            <footer class="c-modal-footer" data-test-id="pie-modal-footer">
+                ${this.renderLeadingAction()}
+                ${this.renderSupportingAction()}
+            </footer>`;
+    }
+
+    /**
+     * Renders the loading spinner if `isLoading` is true.
+     * @private
+     */
+    private renderLoadingSpinner () : TemplateResult | typeof nothing {
+        if (!this.isLoading) {
+            return nothing;
+        }
+
+        return html`<pie-spinner size="xlarge" variant="secondary"></pie-spinner>`;
+    }
+
+    /**
+     * Renders the modal inner content and footer of the modal.
+     * @private
+     */
+    private renderModalContentAndFooter () : TemplateResult {
+        return html`
+            <article class="c-modal-scrollContainer c-modal-content c-modal-content--scrollable">
                 <div class="c-modal-contentInner" data-test-id="modal-content-inner">
                     <slot></slot>
                 </div>
-                ${this.isLoading ? html`<pie-spinner size="xlarge" variant="secondary"></pie-spinner>` : nothing}
+                ${this.renderLoadingSpinner()}
             </article>
-            ${hasFooterLeadingAction ? html`
-                <footer class="c-modal-footer" data-test-id="pie-modal-footer">
-                    ${this.renderLeadingAction()}
-                    ${this.renderSupportingAction()}
-                </footer>` : nothing}`;
+            ${this.renderModalFooter()}`;
     }
 
     public render () {
-        const {
-            aria,
-            hasBackButton,
-            hasStackedActions,
-            heading,
-            headingLevel,
-            isDismissible,
-            isFooterPinned,
-            isFullWidthBelowMid,
-            isLoading,
-            leadingAction,
-            position,
-            size,
-            supportingAction,
-        } = this;
-
-        const headingTag = unsafeStatic(headingLevel);
+        const headingTag = unsafeStatic(this.headingLevel);
+        const ariaLabel = (this.isLoading && this.aria?.loading) || undefined;
+
+        const modalClasses = {
+            'c-modal': true,
+            [`c-modal--${this.size}`]: true,
+            'c-modal--top': this.position === 'top',
```

Comment:
`position: centre` doesn't really do anything, so I'm only applying the class conditionally. I could probably do the same with the size, only applying it for small or large, thoughts?